### ISSUE TITLE
Stop-gradient

### ISSUE THREAD

**Issue by jdk9405** [lpiccinelli-eth/UniK3D#35]
Hi, thanks for sharing the amazing work! 

I have a question about the "Gradient Detach"  in the paper Fig. 2
While reading the repository, I could not find where the proposed detach is implemented. Is it meant to be the following lines?
```
logradius, logconfidence, lowres_features = self.radial_module(
    features=features,
    rays_hr=rays.detach(),
    pos_embed=pos_embed,
    level_embed=level_embed,
)
```
from ```decoder.py``` at [line 490](https://github.com/lpiccinelli-eth/UniK3D/blob/main/unik3d/models/decoder.py#L490).

</br>
</br>

I am also curious whether the following block is used during training or if it is only kept for reference:
```
### LEGACY CODE for training
# if self.training:
#     prob = 1 - tanh(self.steps / self.num_steps)
#     where_use_gt_rays = torch.rand(B, 1, 1, device=device, dtype=dtype) < prob
#     where_use_gt_rays = where_use_gt_rays.int()
#     rays = rays_gt * where_use_gt_rays + rays_pred * (1 - where_use_gt_rays)
```
from ```decoder.py``` at [line 443](https://github.com/lpiccinelli-eth/UniK3D/blob/main/unik3d/models/decoder.py#L432C1-L437C87).
